From 6bb8e6d4c38f260109852ef8b43876ea1aff3810 Mon Sep 17 00:00:00 2001
From: Werner Fink <werner@suse.de>
Date: Tue, 22 Mar 2016 10:38:59 +0100
Subject: [PATCH 165/173] sulogin: Always make echo work after performing
 getpasswd even if root account is locked

If the root account is locked and no password was provided then the terminal
line is not set back to do echo of the input. This correct a small overlook
in commit 7ff1162e67164cb4ece19dd809c26272461aa254

Upstream: http://github.com/karelzak/util-linux/commit/6988998b66b4b95c494d60599f8e3de2ffbaeece
Addresses: http://bugzilla.redhat.com/show_bug.cgi?id=1561200
Signed-off-by: Werner Fink <werner@suse.de>
---
 login-utils/sulogin.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/login-utils/sulogin.c b/login-utils/sulogin.c
index a17b91d71..a6918282e 100644
--- a/login-utils/sulogin.c
+++ b/login-utils/sulogin.c
@@ -676,8 +676,7 @@ quit:
 	alarm(0);
 	if (tc)
 		tcsetattr(fd, TCSAFLUSH, &con->tio);
-	if (ret && *ret != '\0')
-		tcfinal(con);
+	tcfinal(con);
 	printf("\r\n");
 out:
 	return ret;
-- 
2.14.4

